src/test/java/bio/terra/service/filedata/azure/tables/TableFileDaoTest.java

```diff
@@ -1,105 +1,119 @@
   private static final String FILE_ID = UUID.randomUUID().toString();
-  private final TableEntity entity =
-      new TableEntity(PARTITION_KEY, FILE_ID)
-          .addProperty(FireStoreFile.FILE_ID_FIELD_NAME, FILE_ID)
-          .addProperty(FireStoreFile.MIME_TYPE_FIELD_NAME, "application/json")
-          .addProperty(FireStoreFile.DESCRIPTION_FIELD_NAME, "A test entity")
-          .addProperty(FireStoreFile.BUCKET_RESOURCE_ID_FIELD_NAME, "bucketResourceId")
-          .addProperty(FireStoreFile.LOAD_TAG_FIELD_NAME, "loadTag")
-          .addProperty(FireStoreFile.FILE_CREATED_DATE_FIELD_NAME, "fileCreatedDate")
-          .addProperty(FireStoreFile.GS_PATH_FIELD_NAME, "gsPath")
-          .addProperty(FireStoreFile.CHECKSUM_CRC32C_FIELD_NAME, "checksumCrc32c")
-          .addProperty(FireStoreFile.CHECKSUM_MD5_FIELD_NAME, "checksumMd5")
-          .addProperty(FireStoreFile.SIZE_FIELD_NAME, 1L);
-
-  @MockBean private AzureAuthService authService;
-  @MockBean private TableServiceClient tableServiceClient;
-  @MockBean private TableClient tableClient;
-  @MockBean private FireStoreUtils fireStoreUtils;
-
-  @MockBean
-  @Qualifier("performanceThreadpool")
-  private ExecutorService executor;
 
-  @Autowired private TableFileDao dao;
-
-  @Before
-  public void setUp() throws Exception {
-    dao = spy(dao);
-    when(authService.getTableServiceClient(any(), any(), any())).thenReturn(tableServiceClient);
-    when(tableServiceClient.getTableClient(any())).thenReturn(tableClient);
-    when(tableClient.getEntity(PARTITION_KEY, FILE_ID)).thenReturn(entity);
-    when(tableClient.getEntity(PARTITION_KEY, "nonexistentFile"))
-        .thenThrow(TableServiceException.class);
+  private TableEntity createTableEntity(boolean hasMimeType) {
+    TableEntity entity =
+        new TableEntity(PARTITION_KEY, FILE_ID)
+            .addProperty(FireStoreFile.FILE_ID_FIELD_NAME, FILE_ID)
+            .addProperty(FireStoreFile.DESCRIPTION_FIELD_NAME, "A test entity")
+            .addProperty(FireStoreFile.BUCKET_RESOURCE_ID_FIELD_NAME, "bucketResourceId")
+            .addProperty(FireStoreFile.LOAD_TAG_FIELD_NAME, "loadTag")
+            .addProperty(FireStoreFile.FILE_CREATED_DATE_FIELD_NAME, "fileCreatedDate")
+            .addProperty(FireStoreFile.GS_PATH_FIELD_NAME, "gsPath")
+            .addProperty(FireStoreFile.CHECKSUM_CRC32C_FIELD_NAME, "checksumCrc32c")
+            .addProperty(FireStoreFile.CHECKSUM_MD5_FIELD_NAME, "checksumMd5")
+            .addProperty(FireStoreFile.SIZE_FIELD_NAME, 1L);
+    if (hasMimeType) {
+      entity.addProperty(FireStoreFile.MIME_TYPE_FIELD_NAME, "application/json");
+    }
+    return entity;
   }
 
   @Test
-  public void testRetrieveFileMetadata() {
-    FireStoreFile fileMetadata = dao.retrieveFileMetadata(tableServiceClient, DATASET_ID, FILE_ID);
+  void testFromTableEntityMissingMimeType() {
+    TableEntity entity = createTableEntity(false);
```

Comment:
Maybe this should be in its own class, e.g., `FireStoreFileTest`?

Reply:
Good point! Then I can get rid of the nested class